addons/tmi/services/oauth.gd

```diff
@@ -188,7 +188,10 @@ func _send_response(status_code: String, body: String):
 	peer.disconnect_from_host()
 	peer = null
 	
-func _lookup_channel(access_token):
+func _lookup_channel(access_token, channel = null):
+	if (channel == null and credentials != null and !credentials.channel.is_empty()):
+		channel = credentials.channel
+
```

Comment:
if channel is still null or empty at this point, we should probably throw a validation error.  

This can either be done with an assert, or simply doing a push_error message and returning.

Reply:
The issue here is that at the time _lookup_channel is called by _code_to_token, the new credentials haven't been set. _lookup_channel is called in the process of setting up the credentials. 

I made the channel parameter optional and defaulted to credentials.channel if it's set because it seems to work once we're using _idtoken_credentials, but as it was implemented, it was causing crashes for Breadbotnik who was trying to use it for the first time and running into errors from it expecting this value to be set when it couldn't have been yet.

Reply:
I like having channel as a parameter, but I don't like computing a default in the function here.

Instead of allowing nulls, the places where this is called can be guarded to always supply a valid channel name.  Your change in `code_to_token` is already doing that, which means it'll never hit this code path.  Meanwhile `_idtoken_credentials` could just pass in the channel name from the existing credentials like it's already doing to assign a channel name to the new credentials.

Reply:
Yeah, that makes sense. I've fixed this and removed the ternary operators I'd added below to make the code more readable.